boards/arm/atsame54_xpro/atsame54_xpro.dts

```diff
@@ -121,18 +121,19 @@ zephyr_udc0: &usb0 {
 	pinctrl-names = "default";
 
 	mac-eeprom = <&eeprom>;
-
-	phy: phy {
-		compatible = "ethernet-phy";
-		status = "okay";
-		address = <0>;
-		mdio = <&mdio>;
-	};
+	phy-dev = <&phy>;
 };
 
 &mdio {
 	status = "okay";
 
 	pinctrl-0 = <&mdio_default>;
 	pinctrl-names = "default";
+
+	phy: phy {
+		compatible = "ethernet-phy";
+		status = "okay";
+		address = <0>;
+		mdio = <&mdio>;
```

Comment:
The line 137 create a circular dependency now and should be dropped at ethernet-phy.yaml. Since there will be the proper init sequence the generic driver should reflect that.

Reply:
Well it does not create a circular dependency cause the dependency works in the other direction: this says that this node depends on its parent, which is already implied by the hierarcy, so it's redundant. So yeah I know that and I'm well determined to drop the property and use a `DT_PARENT` in the driver, but first I need to fix the hierarchy on another couple of driver. You mind if I do it as a followup? Otherwise I'll have to increse the scope of this to both fix the priority and refactor the binding and hierarchy.